app/server/api/disbursement_api.py

```diff
@@ -0,0 +1,203 @@
+            'status': 'success',
+            'task_uuid': task_uuid,
+            'disbursement_id': d.id,
+            'recipient_count': len(transfer_accounts),
+            'total_disbursement_amount': disbursement_amount*len(transfer_accounts)
+        }
+        return make_response(jsonify(response_object)), 201
+
+
+class DisbursementAPI(MethodView):
+    @requires_auth(allowed_roles={'ADMIN': 'admin'})
+    def get(self, disbursement_id):
+        transfers = db.session.query(CreditTransfer)\
+            .filter(CreditTransfer.disbursement.has(id=disbursement_id))\
+            .options(joinedload(CreditTransfer.disbursement))
+
+        transfers, total_items, total_pages, new_last_fetched = paginate_query(transfers)
+
+        if transfers is None:
+            response_object = {
+                'message': 'No credit transfers',
+            }
+
+            return make_response(jsonify(response_object)), 400
+
+        transfer_list = credit_transfers_schema.dump(transfers).data
+
+
+        response_object = {
+            'status': 'success',
+            'message': 'Successfully Loaded.',
+            'items': total_items,
+            'pages': total_pages,
+            'last_fetched': new_last_fetched,
+            'data': {
+                'credit_transfers': transfer_list,
+            }
+        }
+        return make_response(jsonify(response_object)), 200
+
+    @requires_auth(allowed_roles={'ADMIN': 'admin'})
+    def put(self, disbursement_id):
+        put_data = request.get_json()
+        action = put_data.get('action', '').upper()
+
+        if not disbursement_id:
+            return { 'message': 'Please provide a disbursement_id'}
+        if not action:
+            return { 'message': 'Please provide an action'}
+        if action not in ['EXECUTE']: # We might want more actions later!
+            return { 'message': f'{action} not a valid action' }
+
+        disbursement = Disbursement.query.filter(Disbursement.id == disbursement_id)\
+            .options(joinedload(Disbursement.credit_transfers))\
+            .first()
+        # Since credit transfers are created async, and transfer accounts are added synchronously, we can 
+        # check that it's ready to execute by comparing these lists!
+        if len(disbursement.transfer_accounts) > len(disbursement.credit_transfers):
+            return { 'message': f'Please wait for disbursement creation to complete' }
+
+        if not disbursement:
+            return { 'message': f'Disbursement with ID \'{disbursement_id}\' not found' }
+
+        if disbursement.is_executed == True:
+            return { 'message': f'Disbursement with ID \'{disbursement_id}\' has already been executed!' }
+
+        @status_checkable_executor_job
+        def trigger_jobs(transfers):
+            from server.models.credit_transfer import CreditTransfer
+            # Disabled batch_uuid, since executing two sequential bulk disbursements is unacceptably slow
```

Comment:
Just making a note on this! I realize being slow might not be a dealbreaker since it's an async task, but it's _really really_ slow.